### PR TITLE
Compatibility fix for precice v3 breaking changes

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Fix compatibility esp. with respect to changes from https://github.com/precice/precice/pull/1662 and https://github.com/precice/precice/pull/1742

### CODE REVIEW

File: sample/precice-3-config.xml [hunk 1]
```diff
@@ -1,240 +1,238 @@
...
+    <data:vector name="Moments_0_10"/>
+    <data:vector name="Forces_0_11"/>
+    <data:vector name="Moments_0_11"/>
+    <data:vector name="Forces_0_12"/>
+    <data:vector name="Moments_0_12"/>
+
+    <data:scalar name="Azimuths0"/>
+    <data:scalar name="Collocations0"/>
+    <data:scalar name="Motions0"/>
+    <mesh name="Fluid_Collocations" dimensions="3" >
```
Comment (BenjaminRodenberg) [inline sample/precice-3-config.xml:33]: Also added the `dimensions="3"` in all `mesh` tags. This is easy to oversee in the diff. Do you want to visualize this as well? I'm not sure about the concrete plans here.

Comment (fsimonis) [reply sample/precice-3-config.xml:33]: Could be useful

**Comment by uekerman** [comment]
This is not breaking, right? Meaning the current `master` branch now supports v2 and v3 configs?

File: bin/precice-config-visualizer [hunk 2]
```diff
@@ -66,9 +66,19 @@ def getParticipantColor(solverinterface, blackOnly=False):
...
+
+    if(len(solverinterfaces) == 1):
+        precice_version = 2
+    elif(len(solverinterfaces) == 0):
+        precice_version = 3
+
+    if(precice_version == 2):
+        root = solverinterfaces[0]
+    elif(precice_version == 3):
+        root = ast
```
Comment (BenjaminRodenberg) [inline bin/precice-config-visualizer:81]: The compatibility question is basically handled here. @uekerman do you think we need some additional comments to make this more understandable? I think the code is fairly readable.

Comment (uekerman) [reply bin/precice-config-visualizer:81]: This answers my question: It does support v2 and v3.
We could probably also make this explicit in the documentation. Maybe some users wonder as well.

**Pull request merged**